### PR TITLE
feat(hooks): add useMenu

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

implements useMenu

**Result**

Menu has been added, only main difference with React InstantSearch (v6) is that it doesn't have searchable

FX-442

### CODE REVIEW

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:493]: In other hooks, we've documented the default values for optional arguments with the [`@default`](https://jsdoc.app/tags-default.html) directive, at least for primitives (see [`useHits`](https://github.com/algolia/react-instantsearch/tree/feat/hooks-menu/packages/react-instantsearch-hooks#usehits)).

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:486]:
```suggestion
   * The attribute name in the Algolia records.
```

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:491]:
```suggestion
   * The max number of items to display when `showMoreLimit` is `false`
   * or when the widget is showing fewer values.
```

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:500]:
```suggestion
   * The max number of items to display if the widget is showing more items.
```

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:504]: I'm not sure it's designed for values, but could we leverage the [`@enum`](https://jsdoc.app/tags-enum.html) directive here?

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:506]:
```suggestion
   * You can also pass a sort function that behaves like the standard JavaScript [`compareFunction`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#Syntax).
```

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:508]: Is there any documentation on the `facetOrdering` setting? Couldn't find it in the main docs.

```suggestion
   * If a `facetOrdering` is set in the index settings, it's used for sorting
   * unless you specify `sortBy`.
```

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:526]:
```suggestion
   * The value of the menu item.
```

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:530]:
```suggestion
   * Human-readable value of the menu item.
```

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:534]:
```suggestion
   * Number of matched results after a refinement is applied.
```

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:538]:
```suggestion
   * Whether the list item is refined or not.
```

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:549]:
```suggestion
   * Creates the next state URL for a selected refinement.
```

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:553]:
```suggestion
   * Function to apply selected refinements.
```

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:557]:
```suggestion
   * Send event to Insights middleware
```

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:565]:
```suggestion
   * Whether a refinement can be applied.
```

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:570]:
```suggestion
   * Whether there are enough additional items to display over the `limit`.
```

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:574]:
```suggestion
   * Whether the menu is displaying all the menu items.
```

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [inline packages/react-instantsearch-hooks/README.md:578]:
```suggestion
   * Toggles the number of displayed values between `limit` and `showMoreLimit`.
```

File: examples/hooks/App.tsx
Comment (Haroenv) [reply examples/hooks/App.tsx:57]: makes sense, will do in a followup

File: packages/react-instantsearch-hooks/README.md
Comment (Haroenv) [reply packages/react-instantsearch-hooks/README.md:486]: this is copied from InstantSearch.js, so should be changed there instead

File: packages/react-instantsearch-hooks/README.md
Comment (Haroenv) [reply packages/react-instantsearch-hooks/README.md:491]: https://github.com/algolia/react-instantsearch/pull/3197#discussion_r754108693

File: packages/react-instantsearch-hooks/README.md
Comment (Haroenv) [reply packages/react-instantsearch-hooks/README.md:504]: typescript is already sufficient for that no?

**Comment by Haroenv** [comment]
Thanks for the review of the jsDoc, but I don't think this is the right place to review them, as it's copied from InstantSearch.js. Do you think it makes more sense to make a PR changing the documentation there (and also for all widgets, as they are fairly consistent ATM). Thanks @sarahdayan

File: examples/hooks/App.tsx
Comment (Haroenv) [reply examples/hooks/App.tsx:57]: https://github.com/algolia/react-instantsearch/pull/3203

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [reply packages/react-instantsearch-hooks/README.md:486]: Ah, too bad.

File: packages/react-instantsearch-hooks/README.md
Comment (sarahdayan) [reply packages/react-instantsearch-hooks/README.md:504]: I was mentioning it because you documented it in the JSDoc although it's already in the types. But after checking closer, `@enum` isn't really designed for that.

**Comment by sarahdayan** [comment]
>Do you think it makes more sense to make a PR changing the documentation there

Yes it could be nice to do a pass on the InstantSearch.js ones and these ones. However if we have to make it a whole task, this isn't really a big priority. Maybe we can wait until we tackle the revamp of our API docs (likely in Q1)?

File: examples/hooks/App.tsx
Comment (sarahdayan) [reply examples/hooks/App.tsx:57]: Approved ✅ Once it's merged will you add it for this one?

File: examples/hooks/components/Menu.tsx
Comment (francoischalifour) [inline examples/hooks/components/Menu.tsx:31]: We don't need to prevent the default browser behavior? (and let special events do their behaviors)

Comment (Haroenv) [reply examples/hooks/components/Menu.tsx:31]: good call, I changed last second from inputs to a and forgot to prevent default

Comment (Haroenv) [reply examples/hooks/components/Menu.tsx]: done in https://github.com/algolia/react-instantsearch/pull/3197/commits/7d9978e67e755714b4e0d6bfdd9349a27f2ae22a#diff-6e1456b9e02009184c4ebc4b1fa6983a8b817d376884f200b2df6edc74fe87caR33-R36 now

File: examples/hooks/App.tsx
Comment (tkrugg) [inline examples/hooks/App.tsx:62]: just `showMore` without `{true}` is enough no?
it's idiomatic enough in React that I don't worry about its readability.

File: packages/react-instantsearch-hooks/README.md
Comment (tkrugg) [inline packages/react-instantsearch-hooks/README.md:513]: I'm wondering how helpful theses types are, written as is. 
I think seeing `(items: MenuItem[]) => MenuItem[]` would be more explicit, for documentation.
WDYT?

File: packages/react-instantsearch-hooks/README.md
Comment (tkrugg) [inline packages/react-instantsearch-hooks/README.md:589]: I would show all the parameters here, not just items.
I find that makes the example easier to copy-paste if not working with typescript.

**Review by tkrugg — APPROVED** [review]

File: examples/hooks/App.tsx
Comment (Haroenv) [reply examples/hooks/App.tsx:62]: it seemed consistent with the other usages, can change after these are merged?

File: packages/react-instantsearch-hooks/README.md
Comment (Haroenv) [reply packages/react-instantsearch-hooks/README.md:589]: can do that after on all already made examples?

File: packages/react-instantsearch-hooks/README.md
Comment (Haroenv) [reply packages/react-instantsearch-hooks/README.md:513]: good point, in some others this is "made explicit"

**Review by sarahdayan — APPROVED** [review]

**Pull request merged**
